test/test-utils.mjs

```diff
@@ -0,0 +1,103 @@
+import {promises as fs} from 'fs';
+import url from 'url';
+import assert from 'assert';
+import tap from 'tap';
+
+export async function pathToRawSource(absPath) {
+  let fileURL = url.pathToFileURL(absPath);
+  return fs.readFile(new URL(fileURL), 'utf8');
+}
+
+// Tests exports that all Pre-Processors should have
+export function testPreProcessorExports(preprocessor, options) {
+  tap.has(
+    preprocessor,
+    {
+      sourceExtensionTypes: Array,
+    },
+    'Exports sourceExtensionTypes'
+  );
+
+  tap.has(
+    preprocessor,
+    {
+      outputExtensionTypes: Array,
+    },
+    'Exports outputExtensionTypes'
+  );
+
+  tap.has(
+    preprocessor,
+    {
+      getPreProcessor: Function,
+    },
+    'Exports getPreProcessor'
+  );
+
+  let preprocessorInstance = preprocessor.getPreProcessor(options);
+  tap.has(
+    preprocessorInstance,
+    {
+      process: Function,
+    },
+    'Exports process()'
+  );
+
+  return;
+}
+
+/**
+ * Takes input files and asserts that they match output files after
+ * being processed by the "processor"
+ *
+ * @param processor A Pre- or Post- Processor object
+ * @param options Options passed to the "process", see loaderconfig.mjs
+ * @param inputs List of input source files to process
+ */
+export async function batchTest(processor, options, inputs) {
+  options = Array.isArray(options)
+    ? options
+    : Array(inputs.length).fill(options);
+
+  let numTests = inputs.length;
```

Comment:
Another style nit - not just here, but wherever, it's probably good to be in the habit of using `const` instead of `let` if the values won't change.

Reply:
Good tip, I'll update accordingly

Reply:
Fixed this everywhere